src/runtime_src/core/tools/xbmgmt2/flash/flasher.cpp

```diff
@@ -40,36 +40,38 @@ Flasher::E_FlasherType Flasher::getFlashType(std::string typeStr)
 {
     std::string err;
     E_FlasherType type = E_FlasherType::UNKNOWN;
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::f_flash_type>(m_device);
-    if (typeStr.empty())
-      typeStr = xrt_core::device_query<xrt_core::query::flash_type>(m_device);
 
-    if (typeStr.empty())
-    {
-        getProgrammingTypeFromDeviceName(mFRHeader.VBNVName, type);
-    }
-    else if (typeStr.compare("spi") == 0)
-    {
+    // check various locations for flash_type 
+    try {
+        if (typeStr.empty())
+            typeStr = xrt_core::device_query<xrt_core::query::f_flash_type>(m_device);
+    } catch (...) {}
+    try {
+        if (typeStr.empty())
+          typeStr = xrt_core::device_query<xrt_core::query::flash_type>(m_device);
+    } catch (...) {}
+    try {
+        if (typeStr.empty())
+            getProgrammingTypeFromDeviceName(mFRHeader.VBNVName, type);
+    } catch (...) {}
+    
+    //map flash_tyoe str to E_FlasherType
+    if (typeStr.compare("spi") == 0) {
```

Comment:
This code is converting a string value to an enumeration value, which could be useful elsewhere in the code.   Please do refactor to code into a helping function.   Something similar to `E_FlasherType getFlasherType(const string & typeStr);`